crud/storage_info.lua

```diff
@@ -0,0 +1,117 @@
+local checks = require('checks')
+local errors = require('errors')
+local fiber = require('fiber')
+local log = require('log')
+
+local const = require('crud.common.const')
+local utils = require('crud.common.utils')
+
+local StorageInfoError = errors.new_class('StorageInfoError')
+
+local storage_info = {}
+
+local STORAGE_INFO_FUNC_NAME = 'storage_info_on_storage'
+local CRUD_STORAGE_INFO_FUNC_NAME = utils.get_storage_call(STORAGE_INFO_FUNC_NAME)
+
+--- Storage status information.
+--
+-- @function storage_info_on_storage
+--
+-- @return a table with storage status.
+local function storage_info_on_storage()
+    return {status = "running"}
+end
+
+storage_info.storage_api = {[STORAGE_INFO_FUNC_NAME] = storage_info_on_storage}
+
+--- Polls replicas for storage state
+--
+-- @function call
+--
+-- @tparam ?number opts.timeout
+--  Function call timeout
+--
+-- @tparam ?string|table opts.vshard_router
+--  Cartridge vshard group name or vshard router instance.
+--
+-- @return a table of storage states by replica id.
+function storage_info.call(opts)
+    checks({
+        timeout = '?number',
+        vshard_router = '?string|table',
+    })
+
+    opts = opts or {}
+
+    local vshard_router, err = utils.get_vshard_router_instance(opts.vshard_router)
+    if err ~= nil then
+        return nil, StorageInfoError:new(err)
+    end
+
+    local replicasets, err = vshard_router:routeall()
+    if replicasets == nil then
+        return nil, StorageInfoError:new("Failed to get router replicasets: %s", err.err)
+    end
+
+    local futures_by_replicas = {}
+    local replica_state_by_id = {}
+    local async_opts = {is_async = true}
+    local timeout = opts.timeout or const.DEFAULT_VSHARD_CALL_TIMEOUT
+
+    for _, replicaset in pairs(replicasets) do
+        for replica_id, replica in pairs(replicaset.replicas) do
+            local master = utils.get_replicaset_master(replicaset, {cached = false})
+
```

Comment:
What is there an idea to get master on each iteration?

Reply:
It seems that's one more thing we had missed on the original PR review. Updated. This commit only moves existing code, so I had put improvements to a separate commit "storage: refactor replicasets process on info".